version/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+# Generate a O2Version.h header that contains a couple of useful
+# fields such as the git commit with which we build.
+# We can also use this to define VERSION MACROS etc.
+
+if(DEFINED ENV{O2_GIT_REVISION})
+set(O2_GIT_REVISION $ENV{O2_GIT_REVISION})
+else()
+set(O2_GIT_REVISION "--unknown--")
+endif()
+
+if(DEFINED ENV{ALIDIST_GIT_REVISION})
+set(ALIDIST_GIT_REVISION $ENV{ALIDIST_GIT_REVISION})
+else()
+set(ALIDIST_GIT_REVISION "--unknown--")
+endif()
+
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/O2Version.h.in ${CMAKE_CURRENT_BINARY_DIR}/O2Version.h @ONLY)
```

Comment:
Perhaps we also want to encode whether the build folder was clean, or whether there were local changes?

Reply:
This can be done from the outside. I am preparing a parallel PR for alidit. There we can set O2_GIT_REVISION="abc123 (+unmerged changes)" or just "abc123" when it is clean.

Reply:
OK, I see, of course this depends on some alidist magic then. It will not work if `make install` is used directly, but perhaps it doesn't need to. One could also just fetch the current commit using `git` in that case, but again, probably not necessary.

Reply:
Not it does no longer depend on changes in alidist.